From e0ea139ba13ea81f781079540d236f9b403774c5 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 19 Nov 2012 16:19:42 +0900
Subject: [PATCH 22/39] util: don't skip waiting for state change completion
 in error condition

REL OMXIL component can still safely transition into Idle and Loaded
states after an error occurs.
The client should wait for state change completion even when the
component returns an error.
---
 omx/gstomx_util.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/omx/gstomx_util.c b/omx/gstomx_util.c
index 1175391..d76f282 100644
--- a/omx/gstomx_util.c
+++ b/omx/gstomx_util.c
@@ -748,9 +748,6 @@ wait_for_state (GOmxCore * core, OMX_STATETYPE state)
 
   g_mutex_lock (core->omx_state_mutex);
 
-  if (core->omx_error != OMX_ErrorNone)
-    goto leave;
-
   g_get_current_time (&tv);
   g_time_val_add (&tv, 15 * G_USEC_PER_SEC);
 
-- 
1.7.9.5

